### PR TITLE
[180] Auth Context

### PR DESCRIPTION
**Pull request by katelovescode**
# 💅🏼 What issue does this fix?
Related to/came out of #180 but doesn't finish the actual point of that ticket.  This just moves localStorage calls to a use-local-storage-state hook, and moves all the auth functionality into a context to be shared with other components.

## 🍂 Before You Submit

* [ ] Did you write tests?
* [X] Did you run `bundle exec rspec` from the root?
* [X] Did you run `bundle exec rails rswag` from the root?
* [X] Did you run `bundle exec rubocop` from the root?
* [X] Did you run `yarn lint` in `/client`?
* [X] Did you run `yarn test` in `/client`?
* [ ] Are your primary keys UUIDs on any new tables?

## 🛷 Deployment Considerations

* [ ] Data Migrations
* [ ] Schema Migrations
* [X] Dependencies

## 🧵 Steps to set up locally

`yarn --cwd client install`

## 🧳 Steps to test
Check all the authorized routes and non-authorized routes and make sure it's working as expected

## 🕯 Caveats, concerns
Still trying to figure out the expiration stuff; this has expiration in the context, but it's not used, it's a placeholder for expiration behavior if we can get it to work.

### CODE REVIEW

File: client/src/i18n/locales/es/index.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
   "resendConfirmationEmail": "Haga clic aquí para reenviar el correo.",
   "toResendConfirmationEmail": "para reenviar el correo de confirmación.",
   "confirmationEmailResent": "Correo reenviado",
+  "confirmYourAccount": "Confirme tu cuenta",
```
Comment (katelovescode) [inline client/src/i18n/locales/es/index.json:66]: @csprayregen please confirm translation

File: client/src/_utils/_routes/AuthenticatedRoute.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  path,
+  title,
+  // permissions,
+  ...routeProps
+}) {
+  exact = !!exact
+  let history = useHistory()
+  const authenticated = IsAuthenticated
+
+  useEffect(() => {
```
Comment (DoodlingDev) [inline client/src/_utils/_routes/AuthenticatedRoute.js:20]: I'm curious about this `useEffect` here. Without a dependency array, does it run every time the function is re-rendered? So it's the same as not being in a use Effect?

Comment (katelovescode) [reply client/src/_utils/_routes/AuthenticatedRoute.js:20]: Good point, it's probably not necessary to do it in the useEffect, gonna test that theory.

**Comment by katelovescode** [comment]
Closing this in favor of redux

**Pull request closed without merging**